src/compositor/lipstickcompositor.cpp

```diff
@@ -700,9 +702,12 @@ void LipstickCompositor::setUpdatesEnabled(bool enabled)
                 m_onUpdatesDisabledUnfocusedWindowId = topmostWindow->windowId();
                 clearKeyboardFocus();
             }
-            hide();
-            if (QWindow::handle()) {
-                QGuiApplication::platformNativeInterface()->nativeResourceForIntegration("DisplayOff");
+            if (calledFromDBus()) {
+                setDelayedReply(true);
+                m_displayOffReply = message().createReply();
+                startTimer(1000);
```

Comment:
@spiiroin correct me if this is bs, but IIRC mce won't ask for setUpdatesEnabled(true) until the previous request has completed, so this will introduce a delay for switching on the display right after it went off. Not that critical, maybe?